### PR TITLE
Creating a family or collection now used the organisation of the user

### PR DESCRIPTION
**Pull request by diversemix**
# Description

- Any family or collection that is created has its organization set to that of the logged in user 
- This updates the "CreateDTO" /cc @PatFawbertMills 
- Tests updated

## Type of change

Please select the option(s) below that are most relevant:

- [x] Bug fix
- [ ] New feature
- [ ] Breaking change

## How Has This Been Tested?

Please describe the tests that you added to verify your changes.

## Reviewer Checklist

- [ ] The PR represents a single feature (small driveby fixes are also ok)
- [ ] The PR includes tests that are sufficient for the level of risk
- [ ] The code is sufficiently commented, particularly in hard-to-understand areas
- [ ] Any required documentation updates have been made
- [ ] Any TODOs added are captured in future tickets
- [ ] No FIXMEs remain

### CODE REVIEW

File: app/model/family.py [hunk 1]
```diff
@@ -24,25 +24,34 @@ class FamilyReadDTO(BaseModel):
...
+    """
+    A JSON representation of a family for creating.
+
+    Note:
+     - import_id is auto generated
+     - slug is auto generated
+     - organisation comes from the user's organisation
+    """
 
     # import_id: not included as generated
```
Comment (katybaulch) [inline app/model/family.py:52]: Can we remove this line as we've now added to the docstring that the import_id is auto generated?

File: app/repository/helpers.py [hunk 2]
```diff
@@ -61,7 +61,9 @@ def generate_import_id(
     if type(org) == str:
         org_name = org
     else:
-        org_name = db.query(Organisation.name).filter(Organisation.id == org)
+        org_name = (
+            db.query(Organisation.name).filter(Organisation.id == org).scalar_subquery()
```
Comment (katybaulch) [inline app/repository/helpers.py:65]: Nice!

File: unit_tests/mocks/services/collection_service.py [hunk 3]
```diff
@@ -38,7 +38,9 @@ def mock_update_collection(
             return create_collection_read_dto(import_id, data.title, data.description)
 
     # TODO: Think: Is this return value correct?
```
Comment (katybaulch) [inline unit_tests/mocks/services/collection_service.py:40]: Do we still need this TODO?

File: unit_tests/service/test_collection_service.py [hunk 4]
```diff
@@ -140,31 +139,33 @@ def test_update_raises_when_invalid_id(
...
+    app_user_repo_mock,
 ):
     new_collection = create_dto(import_id="a.b.c.d")
     collection_repo_mock.return_empty = True
     with pytest.raises(RepositoryError) as e:
-        collection_service.create(_to_create_dto(new_collection))
+        collection_service.create(_to_create_dto(new_collection), USER_EMAIL)
     assert e.value.message == "Error when creating collection 'description'"
     assert collection_repo_mock.create.call_count == 1
     # Ensure the collection service uses the geo service to validate
```
Comment (katybaulch) [inline unit_tests/service/test_collection_service.py:167]: Cut paste comment needs updating to say collections service uses the app user repo to validate.

File: unit_tests/service/test_family_service.py [hunk 5]
```diff
@@ -226,46 +228,47 @@ def test_update_family_raises_when_metadata_invalid(
...
-    organisation_repo_mock,
-    metadata_repo_mock,
+    family_repo_mock, geography_repo_mock, metadata_repo_mock, app_user_repo_mock
 ):
     new_family = create_family_dto(import_id="A.0.0.5")
-    family = family_service.create(to_create_dto(new_family))
+    family = family_service.create(to_create_dto(new_family), USER_EMAIL)
     assert family is not None
     assert family_repo_mock.create.call_count == 1
     # Ensure the family service uses the geo service to validate
```
Comment (katybaulch) [inline unit_tests/service/test_family_service.py:237]: Should probably either remove this comment, or replace it to be more generic to capture the other services we are validating that the family service is calling.

File: unit_tests/service/test_family_service.py [hunk 6]
```diff
@@ -226,46 +228,47 @@ def test_update_family_raises_when_metadata_invalid(
...
     metadata_repo_mock,
+    app_user_repo_mock,
 ):
     new_family = create_family_dto(import_id="a.b.c.d")
     family_repo_mock.return_empty = True
-    family = family_service.create(to_create_dto(new_family))
+    family = family_service.create(to_create_dto(new_family), USER_EMAIL)
     assert family is False
     assert family_repo_mock.create.call_count == 1
     # Ensure the family service uses the geo service to validate
```
Comment (katybaulch) [inline unit_tests/service/test_family_service.py:254]: Should probably either remove this comment, or replace it to be more generic to capture the other services we are validating that the family service is calling.

**Review by katybaulch — APPROVED** [review]
All good functionally - a couple of little updates to comments ideally needed but nothing stopping a merge :+1:

File: unit_tests/mocks/services/collection_service.py (hunk 3 above)
Comment (diversemix) [reply unit_tests/mocks/services/collection_service.py:40]: I'll fix the todos in https://linear.app/climate-policy-radar/issue/PDCT-520/

File: unit_tests/service/test_collection_service.py (hunk 4 above)
Comment (diversemix) [reply unit_tests/service/test_collection_service.py:167]: Oops - removed the check and not the comment

File: unit_tests/service/test_family_service.py (hunk 6 above)
Comment (diversemix) [reply unit_tests/service/test_family_service.py:254]: Oops - removed the check and not the comment

**Pull request merged**